File: Java/implement-strstr.java
class Solution {
    public int strStr(String haystack, String needle) {
        if (needle.length() == 0) {
            return 0;
        }
        if (haystack.length() == 0 || needle.length() > haystack.length()) {
            return -1;
        }
        int i = 0;
        for (; i <= haystack.length() - needle.length(); i++) {
            boolean isSucceed = true;
            for (int j = 0; j < needle.length(); j++) {
                if (haystack.charAt(i + j) != needle.charAt(j)) {
                    isSucceed = false;
                    break;
                }
            }
            if (isSucceed) {
                return i;
            }
        }
        return -1;
    }
}

File: Kotlin/implement-strstr.kt
class Solution {
    fun strStr(haystack: String, needle: String): Int {
        if (needle.length == 0) return 0
        if (haystack.length < needle.length || haystack.length == 0) return -1
        for (i in 0 until (haystack.length - needle.length)) {
            var isSucceed = true
            for (j in needle.indices) {
                if (haystack[i + j] != needle[j]) {
                    isSucceed = false
                    break
                }
            }
            if (isSucceed) return i
        }
        return -1
    }
}
